stdlib/public/core/Pointer.swift

```diff
@@ -10,11 +10,17 @@
 //
 //===----------------------------------------------------------------------===//
 
+#if SWIFT_ENABLE_REFLECTION
+public protocol _CustomReflectableOrNone: CustomReflectable {}
+#else
+public protocol _CustomReflectableOrNone {}
+#endif
+
 /// A stdlib-internal protocol modeled by the intrinsic pointer types,
 /// UnsafeMutablePointer, UnsafePointer, UnsafeRawPointer,
 /// UnsafeMutableRawPointer, and AutoreleasingUnsafeMutablePointer.
 public protocol _Pointer
-: Hashable, Strideable, CustomDebugStringConvertible, CustomReflectable {
+: Hashable, Strideable, CustomDebugStringConvertible, _CustomReflectableOrNone {
```

Comment:
Can you split out the `CustomReflectable` into an extension instead?  That avoids the need for the new protocol.

Reply:
You can't actually extend protocols to require conformances  to protocols, but couldn't you do:
```swift
public protocol _Pointer: protocols
#if SWIFT_ENABLE_REFLECTION
, CustomReflectable
#endif 
{
}
```
maybe the Swift preprocessor doesn't work like C/++ one to be able to do this.

Reply:
Alas, Swift doesn't do blind text substitutions like C does, and the contents of an `#if` block must be valid statements.

Reply:
Bleh, I missed that `_Pointer` itself was a protocol.

Reply:
How would you use the typealias? I'm happy to explore better options, as I don't really like the `_CustomReflectableOrNone` thing either, but I couldn't come up with anything nicer that would actually work (as mentioned above, we can't just ifdef this out from the declaration, and we can't do extension on a protocol).

Reply:
This appears to work:
```#if COND
typealias MaybeCustomReflectable = CustomReflectable
#else
typealias MaybeCustomReflectable = Any
#endif

protocol P: MaybeCustomReflectable {}
```

Reply:
Yeah, roughly what @mikeash pointed out.  I was thinking that we could take the requirements and just `typealias` them before the definition, including/excluding the `CustomReflectable` instead of an `Any` conformance.